graph/client.py

```diff
@@ -89,16 +93,25 @@ def search_rules_id(self):
         else:
             return rules
 
-    def prepare_graphs(self, rules):
+    def prepare_data(self, rules):
         try:
             for rule in rules['rules']:
-                oval_tree = graph.oval_graph.build_nodes_form_xml(
-                    self.source_filename, rule).to_sigma_dict(0, 0)
-                with open('html_interpreter/data.js', "w+") as graph_data_file:
-                    graph_data_file.write("var data_json =" + str(json.dumps(
-                        oval_tree,
-                        sort_keys=False,
-                        indent=4) + ";"))
+                if self.tree:
+                    oval_tree = graph.oval_graph.build_nodes_form_xml(
+                        self.source_filename, rule).to_JsTree_dict()
+                    with open('tree_html_interpreter/data.js', "w+") as file:
+                        file.write("var data_json =" + str(json.dumps(
+                            oval_tree,
+                            sort_keys=False,
+                            indent=4) + ";"))
+                else:
+                    oval_tree = graph.oval_graph.build_nodes_form_xml(
+                        self.source_filename, rule).to_sigma_dict(0, 0)
+                    with open('graph_html_interpreter/data.js', "w+") as file:
+                        file.write("var data_json =" + str(json.dumps(
+                            oval_tree,
+                            sort_keys=False,
+                            indent=4) + ";"))
```

Comment:
Both if and else branch are very similar. Have you considered writing single code and parametrize it using variables?

Also, this is code is nested quite a lot. (def - try - for - if is the 4th level of indentation)  What about creating a separate function for it?

I'm not explicitly requesting changes, it's quite readable anyway. I would only suggest you thinking a few minutes about it. Feel free to disagree :smile: